From 890a56e3062a2170c9d401cad351f67b01e089c0 Mon Sep 17 00:00:00 2001
From: Zuohai Xu <zouhaix@marvell.com>
Date: Tue, 11 Jun 2019 16:27:56 +0800
Subject: [PATCH 290/386] arm64: dts: marvell: Update dtsi of cn9130 crb

1. Update GPIO pins for SD VCC and VCQ.
2. Add a i2c expander node.
3. Add a signale to control vbus of USB type A socket.
4. Align mpps configuration.

Change-Id: If8ca2d52bab19994b15e9886c4d9328bdf13ce01
Signed-off-by: Zuohai Xu <zouhaix@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10712
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-crb.dtsi | 35 +++++++++++++++++++--
 1 file changed, 32 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-crb.dtsi b/arch/arm64/boot/dts/marvell/cn9130-crb.dtsi
index fd6a25d6b8c8..e5cd35d31707 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-crb.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130-crb.dtsi
@@ -28,12 +28,21 @@
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
 
+	cp0_reg_usb3_vbus1: cp0_usb3_vbus@1 {
+		compatible = "regulator-fixed";
+		regulator-name = "cp0-xhci1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		gpio = <&expander0 8 GPIO_ACTIVE_HIGH>;
+	};
 	cp0_usb3_0_phy0: cp0_usb3_phy0 {
 		compatible = "usb-nop-xceiv";
 	};
 
 	cp0_usb3_0_phy1: cp0_usb3_phy1 {
 		compatible = "usb-nop-xceiv";
+		vcc-supply = <&cp0_reg_usb3_vbus1>;
 	};
 
 	cp0_reg_sd_vccq: cp0_sd_vccq@0 {
@@ -71,14 +80,35 @@
 /*
  * CP0
  */
+
+&cp0_pinctrl {
+	cp0_sdhci_cd_pins_crb: cp0-sdhci-cd-pins-crb {
+		marvell,pins = "mpp55";
+		marvell,function = "gpio";
+	};
+	cp0_spi1_pins_crb: cp0-spi-pins-crb {
+		marvell,pins = "mpp27", "mpp28", "mpp29", "mpp30";
+		marvell,function = "spi1";
+	};
+};
+
 &cp0_i2c0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&cp0_i2c0_pins>;
 	status = "okay";
 	clock-frequency = <100000>;
+	expander0: mcp23x17@20 {
+		compatible = "microchip,mcp23017";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x20>;
+		status = "okay";
+	};
 };
 
 &cp0_i2c1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c1_pins>;
 	clock-frequency = <100000>;
 	status = "okay";
 };
@@ -87,9 +117,8 @@
 &cp0_sdhci0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&cp0_sdhci_pins
-		     &cp0_sdhci_cd_pins>;
+		     &cp0_sdhci_cd_pins_crb>;
 	bus-width = <4>;
-	broken-cd;
 	no-1-8-v;
 	vqmmc-supply = <&cp0_reg_sd_vccq>;
 	vmmc-supply = <&cp0_reg_sd_vcc>;
@@ -98,7 +127,7 @@
 
 &cp0_spi1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&cp0_spi0_pins>;
+	pinctrl-0 = <&cp0_spi1_pins_crb>;
 	reg = <0x700680 0x50>,		/* control */
 	      <0x2000000 0x1000000>;	/* CS0 */
 	status = "okay";
-- 
2.17.1

